### PR TITLE
BaSP-MR-118: Implemented redux logic in form for Classes

### PR DESCRIPTION
**Pull request by fernandofernandezuy**
With the integration of redux in our app we can now effectively manage the state of our application, resulting in smoother data flow, easier tracking of changes, and improved performance.
In this pull request, I implement redux logic for adding or editing a class in this way:
- Move the logic of the post, put it to the thunk file in the classes folder, added the actions and reducer necessaries. 
- Replace the useState to get the data for useSelector

### CODE REVIEW

**Comment by Federico-Peralta** [comment]
[[Classes] - Move post and put logic.](https://trello.com/c/fEPjlSg4/118-classes-move-post-and-put-logic)

File: src/Components/Classes/Form/index.jsx [hunk 1]
```diff
@@ -1,177 +1,121 @@
...
 import Button from '../../Shared/Button';
 import SharedModal from '../../Shared/Modal';
 
 const Form = () => {
-  const history = useHistory();
   const { id } = useParams();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const gymClasses = useSelector((state) => state.classes.data.data);
+  const { trainers = [], activities = [] } = useSelector((state) => ({
```
Comment (marcohzo) [inline src/Components/Classes/Form/index.jsx]:
```suggestion
    const {
    trainers = [],
    activities = [],
    gymClasses = []
  } = useSelector((state) => ({
    trainers: state.trainers.data,
    activities: state.activities.data.data,
    gymClasses: state.classes.data.data
  }));
```

**Review by marcohzo — APPROVED** [review]

**Review by EmaBaldor — COMMENTED** [review]
You can unify the modal component of the class index into one.

File: src/Components/Classes/Form/index.jsx [hunk 2]
```diff
@@ -1,177 +1,127 @@
...
-    try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers`);
-      const data = await response.json();
-      setTrainers(data.data);
-    } catch (error) {
-      console.error('Error fetching trainers:', error);
+  useEffect(() => {
+    dispatch(getTrainers());
+    dispatch(getActivities());
+    if (id) {
```
Comment (EmaBaldor) [inline src/Components/Classes/Form/index.jsx]: A better practice would be to move the conditional logic to a separate function, outside of useEffect.

Comment (fernandofernandezuy) [reply src/Components/Classes/Form/index.jsx]: Good suggestion, thanks

**Review by EmaBaldor — APPROVED** [review]

**Review by EmaBaldor — APPROVED** [review]
Good Job!

**Review by Federico-Peralta — APPROVED** [review]
Good job!

**Comment by nazarenapantanetti** [comment]
Tested in master, changes requested! Follow the link to see the complete report: https://trello.com/c/O4JGoPBW/141-classes-bug-defective-edition

**Comment by nazarenapantanetti** [comment]
Tested and approved! Changes requested were successfully made.

**Pull request merged**